resources/overlay_plugin_api.ts

```diff
@@ -189,10 +196,12 @@ export const init = (): void => {
     return;
 
   if (typeof window !== 'undefined') {
-    wsUrl = /[\?&]OVERLAY_WS=([^&]+)/.exec(window.location.href);
-    if (wsUrl) {
+    wsUrl = urlParams.get('OVERLAY_WS');
+    if (wsUrl !== null) {
       const connectWs = function() {
-        ws = new WebSocket(wsUrl?.[1] as string);
+        if(wsUrl === null)
+          throw new TypeError('wsUrl is null, a string is required');
```

Comment:
```suggestion
```

I don't think `wsUrl` can be null here.

Reply:
I think it can be, when the user didn't run cactbot overlays in ACT/ngld's OverlayPlugin and/or didn't provide the parameter?
```suggestion
        if(wsUrl === null)
          throw new TypeError('Didn\'t found OVERLAY_WS parameter. If this overlay is not running in ACT or ngld\'s OverlayPlugin, make sure you have provided OVERLAY_WS parameter.');
```

Reply:
I agree with everything you said, but there's a `if (wsUrl) {` line above it that prevents the `connectWs` function from even being created if there if `wsUrl` is null.  This is also the only place that `wsUrl` is set.

Reply:
Ah, correct.
Sorry I misreaded it. ಠ︵ಠ

Reply:
by the way, this type checking in is only used to pass the typescript compile test but no any actual usage, because the function is defined in the global namespace so the compiler may think it is dangerous🤣

Reply:
Maybe should be change to something like:
```ts
    wsUrl = urlParams.get('OVERLAY_WS');
    if (wsUrl !== null) {
      const connectWs = function(wsUrl: string): void {
        ws = new WebSocket(wsUrl);

        // ...
        ws.addEventListener('close', () => {
          queue = null;

          console.log('Trying to reconnect...');
          // Don't spam the server with retries.
          window.setTimeout(() => {
            connectWs(wsUrl);
          }, 300);
        });
      };

      connectWs(wsUrl);
    } else {
      // ...
```

Reply:
Also remember to run `npm run lintfix`.